networkx/generators/directed.py

```diff
@@ -277,7 +276,8 @@ def _choose_node(G,distribution,delta):
 
     # seed random number generated (uses None as default)
     random.seed(seed)
-
+    psum_in = G.number_of_edges() + float(delta_in) * len(G)
+    psum_out = G.number_of_edges() + float(delta_out) * len(G)
```

Comment:
I think G is changing so the number of edges will change. 
So I think we will have to compute psum just before we call it.

But maybe faster than ```G.number_of_edges``` we should just count them as we add them.

    count_edges=0
    while ...
        ...
        G.add_edge(u,v)
        count_edges += 1

We could count nodes too, but len(G) does that quickly (G.number_of_edges() has to sum the lengths of 2*len(G) dicts, so its not very fast).